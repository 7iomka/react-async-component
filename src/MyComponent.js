/* @flow */

import React, { PropTypes } from 'react';

// $FlowFixMe - using propTypes
function MyComponent({ msg }) {
  return <h1>{msg}</h1>;
}

MyComponent.propTypes = {
  msg: PropTypes.string.isRequired,
};

export default MyComponent;
