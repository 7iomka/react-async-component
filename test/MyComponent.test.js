/* @flow */

import React from 'react';
import MyComponent from '../src/MyComponent';
import describeComponent from './__helpers__/describeComponent';

describeComponent('MyComponent', () => {
  it('renders', () => {
    expect(<MyComponent msg="foo" />).toMatchSnapshot();
  });

  it('propTypes', () => {
    expect(() => <MyComponent />).toThrowError('Failed prop type');
  });
});
