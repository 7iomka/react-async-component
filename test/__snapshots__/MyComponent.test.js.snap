exports[`MyComponent renders 1`] = `
<MyComponent
  msg="foo" />
`;
